pitop/processing/algorithms/ball_detect.py

```diff
@@ -0,0 +1,247 @@
+from collections import deque
+import numpy as np
+from pitop.core import ImageFunctions
+from pitop.processing.utils.vision_functions import (
+    center_reposition,
+    get_control_angle,
+)
+from typing import Union
+from imutils import resize, grab_contours
+from imutils.video import FPS
+from pitop.core.data_stuctures import DotDict
+import atexit
+from pitop.processing.utils.vision_functions import import_opencv
+
+
+cv2 = import_opencv()
```

Comment:
please check https://github.com/pi-top/pi-top-Python-SDK/blob/master/pitop/camera/core/capture_actions/motion_detector.py#L19 to see how to use `import_opencv` with classes... 

If you run `import_opencv` at the top of a file that's imported whenever you run `import pitop`, it will throw an exception if `OpenCV` is not installed, that's why it was added as a class attribute, so it fails when you actually attempt to use `cv2`... you _could_
 use `cv2` as a global variable, initialized as `None`, and actually run `import_opencv` in the constructor.

On the examples is not that bad, since you know what to expect, and it should fail right away if a library is missing.

Reply:
ok got it